lib/shelley/src/Cardano/Wallet/Shelley.hs

```diff
@@ -351,6 +360,22 @@ serveWallet
                 $ \db@DBLayer{..} -> do
 
             forM_ settings $ atomically . putSettings
+            dbSettings <- atomically readSettings
+            -- initially verity smash server
+            case poolMetadataSource dbSettings of
```

Comment:
This is more or less on application start. We crash the app if the server isn't reachable.